### DISCUSSION TITLE
GET http://localhost:3000/api/auths 500 (Internal Server Error)

### DISCUSSION THREAD

**Discussion by IRediTOTO** [vercel/next.js#12635] [Help]
My code worded very well before. But now it got status 500 :| I can't see any error on console log client or server. 
Is there anyway to know what tool can detect error for nextjs ? Or somewhere i can check for it? :( 
![image](https://user-images.githubusercontent.com/15695252/81407917-742cb480-9166-11ea-90f8-d5f6e9095172.png)

**Comment by zackdotcomputer** [comment]
What errors are you seeing printed in your computer's terminal? (That is to say the server terminal on your computer where you're running next.)

**Reply by IRediTOTO** [reply]
hi, vscode console show nothing, I found error on database setup, but it didnt show error on console (server and browser), I tried `try catch` but still not show. Sometime it make me hours to find what did I do wrong.

**Reply by realraja** [reply]
did you get solution then please tell me

**Reply by Dardabiee** [reply]
> hi, vscode console show nothing, I found error on database setup, but it didnt show error on console (server and browser), I tried `try catch` but still not show. Sometime it make me hours to find what did I do wrong.

have you solved it? i'm getting the same error, please tell me how to fix it

**Reply by zackdotcomputer** [reply]
So a 500 response means a server error. This happens when the server failed to run the code (in which case Next will catch that error and return a 500), or when some code recognized that it has hit an error and chooses to return a 500 because it cannot continue.

You will need to check your own server logs (and possibly add try/catch or console.log statements to debug) in order to figure out what is going wrong. If this is happening on an api route that you created or generated, then you will need to carefully debug that. If this is happening in a library you're using like next-auth, then you should take the issue to their Github or Discord communities for support. Whatever the problem is, I'm afraid to say it is not a NextJS issue and is going to have a different cause and solution for each person who finds this page. There isn't a one-size-fits-all "here's how to fix it" solution.
